yb-voyager/src/query/queryparser/helpers_protomsg.go

```diff
@@ -802,8 +797,74 @@ func TraverseAndExtractDefNamesFromDefElem(msg protoreflect.Message) (map[string
 	return defNamesWithValues, nil
 }
 
-func GetAIndirectionNode(msg protoreflect.Message) (*pg_query.A_Indirection, bool) {
-	protoMsg := msg.Interface().(protoreflect.ProtoMessage)
-	aIndirection, ok := protoMsg.(*pg_query.A_Indirection)
-	return aIndirection, ok
+/*
+Example:
+options:{def_elem:{defname:"security_invoker" arg:{string:{sval:"true"}} defaction:DEFELEM_UNSPEC location:32}}
+options:{def_elem:{defname:"security_barrier" arg:{string:{sval:"false"}} defaction:DEFELEM_UNSPEC location:57}}
+
+Presence-only flag example where arg is absent (nil in parse tree):
+SQL: CREATE TABLE t(id int) WITH (autovacuum_enabled);
+
+ParseTree:	stmt:{create_stmt:{relation:{relname:"t"  inh:true  relpersistence:"p"  location:13}
+
+			table_elts:{column_def:{colname:"id"  type_name:{names:{string:{sval:"pg_catalog"}}  names:{string:{sval:"int4"}} }}
+	    	options:{def_elem:{defname:"autovacuum_enabled"  defaction:DEFELEM_UNSPEC }} }}
+
+NormalizeDefElemArgToString converts common DefElem Arg node shapes to string to avoid warn/noise:
+- nil => "true" (presence-only flags)
+- String_ => sval
+- A_Const => sval/ival/fval/bsval/boolval
+- TypeName => last component of Names (e.g. "icu")
+- List => comma-joined normalized items
+- TypeCast => normalize inner Arg
+- Fallback => arg.String() to keep value non-empty
+*/
+func NormalizeDefElemArgToString(arg *pg_query.Node) string {
+	if arg == nil {
+		return "true"
+	}
+	if s := arg.GetString_(); s != nil {
+		return s.Sval
+	}
+	if a := arg.GetAConst(); a != nil {
+		switch {
+		case a.GetSval() != nil:
+			return a.GetSval().Sval
+		case a.GetIval() != nil:
+			return strconv.FormatInt(int64(a.GetIval().Ival), 10)
+		case a.GetFval() != nil:
+			return a.GetFval().Fval
+		case a.GetBsval() != nil:
+			return a.GetBsval().Bsval
+		case a.GetBoolval() != nil:
+			return a.GetBoolval().String()
+		default:
+			return arg.String()
+		}
+	}
+
+	if i := arg.GetInteger(); i != nil {
+		return strconv.FormatInt(int64(i.Ival), 10)
+	}
+
+	if tn := arg.GetTypeName(); tn != nil {
+		if len(tn.Names) > 0 && tn.Names[len(tn.Names)-1].GetString_() != nil {
+			return tn.Names[len(tn.Names)-1].GetString_().Sval
+		}
+		return arg.String()
+	}
+	if l := arg.GetList(); l != nil {
+		parts := make([]string, 0, len(l.Items))
+		for _, it := range l.Items {
+			v := NormalizeDefElemArgToString(it)
+			if v != "" {
+				parts = append(parts, v)
+			}
+		}
+		return strings.Join(parts, ",")
+	}
+	if tc := arg.GetTypeCast(); tc != nil {
+		return NormalizeDefElemArgToString(tc.Arg)
+	}
+	return arg.String()
```

Comment:
This isn't a complete normalization, right?
As there could be other arg types like expression, function call

Reply:
Are there any other known cases for def elements?
I added what i could find/think of.